### ISSUE TITLE
Optional values returned by LoadRepository.findAll can be null

### ISSUE THREAD

**Issue by kuhnza** [n15g/spring-boot-gae#1]
Calling the default `LoadRepository.findAll` method with a non-existent key (i.e. no entity exists for that key) results in a Map which contains a `null` value for said key. This is unexpected behaviour as the returned Map suggests an optional should always be returned for a queried key. For example:

```java
Key<MyEntity> keyExists = Key.create(MyEntity.class, "legit");
Key<MyEntity> keyDoesNotExist = Key.create(MyEntity.class, "bogus");
Map<Key<MyEntity>, Optional<MyEntity>> resultMap = myRepository.findAll(keyExists, keyDoesNotExist);

assert resultMap.get(keyExists) != null;  // fine, returns optional
assert resultMap.get(keyDoesNotExist) != null;  // oops, I'm null!
```

The current return value _is_ consistent with the underlying Objectify keys method which doesn't contain entries for the non-existent keys.

Two possible options:

1. Scrap the Optional return value. 
2. Change the implementation to stream over the passed collection of keys and ensure there is always an entry regardless of whether it exists or not.

Option 1 is probably safest as without context it's easy for someone to request a value from the map expecting an Optional but instead getting null. They're always free to wrap/map to Optional in their own code if required.

Option 2 avoids a breaking change to the API and might look something like
```java
default Map<Key<E>, Optional<E>> findAll(Collection<Key<E>> keys) {
    Map<Key<E>, E> results = ofy().load().keys(keys);
    return keys.parallelStream()
        .collect(Collectors.toMap(
            key -> key,
            key -> Optional.ofNullable(results.get(key))
        ));
}
```

In any event the current implementation is redundant as all entries in the returned Map will always contain a value.

**Comment by n15g** [comment]
Nice find Dave, cheers.

I opted for Option(al) 2 as I'm doing something similar... somewhere, or maybe I planned to do it here and forgot. Either way, it also doesn't break the API as you point out which is probably best for a patch.

I went for a non-streamy approach as well, as Collectors.toMap has caused me issues elsewhere when you can potentially have a null key, I'm trying to avoid using it where I can: https://stackoverflow.com/questions/24630963/java-8-nullpointerexception-in-collectors-tomap.

I'll release a 1.0.1 version ASAP.

**Comment by n15g** [comment]
@kuhnza as discussed offline we'll go with Option 1 instead. I'll await your PR.

Thanks again.

**Comment by kuhnza** [comment]
Cheers @n15g make a breaking change while it's early and there's little impact makes sense. 

Since we're not intending to use the lazy loading provided by the underlying Objectify Map proxy refactoring the findAll methods to return a List will simplify things and make the interface more consistent.

Will submit a PR.

**Comment by n15g** [comment]
Fixed in 1.1.0-beta-1, awaiting release.
